src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
It would not, but it does solve the current clutter problem at least.  We could eventually make a 'nonPromiseCreator' function as well that handles nonasync bodies